ci-operator/step-registry/mco-day2/install-kernel-type/mco-day2-install-kernel-type-commands.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+
+function validate_params {
+  if  [ "$INSTALL_KERNEL_MCPS" = "" ]
+  then
+    echo "ERROR: No MachineConfigPool provided"
+    exit -1
+  fi
+
+  if  [ "$INSTALL_KERNEL_TYPE" != "realtime" ] && [ "$INSTALL_KERNEL_TYPE" != "64k-pages" ]
+  then
+    echo "ERROR: kerneltype '$INSTALL_KERNEL_TYPE' is not allowed. Accepted values: [realtime, 64k-pages]"
+    exit -1
+  fi
+}
+
+function create_machine_configs {
+  local KERNEL_MCPS=$1
+  local KERNEL_TYPE=$2
+
+  for MACHINE_CONFIG_POOL in $KERNEL_MCPS
+  do
+  MC_NAME="99-$MACHINE_CONFIG_POOL-$KERNEL_TYPE"
+
+  echo "Creating $KERNEL_TYPE kerneltype MachineConfig $MC_NAME for pool $MACHINE_CONFIG_POOL"
+
+  cat << EOF | oc create -f -
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: $MACHINE_CONFIG_POOL
+  name: $MC_NAME
+spec:
+  kernelType: $KERNEL_TYPE
+EOF
+  done
+
+}
+
+function wait_for_config_to_be_applied {
+  local KERNEL_MCPS=$1
+  local KERNEL_TIMEOUT=$2
+
+
+
+  for MACHINE_CONFIG_POOL in $KERNEL_MCPS
+  do
+  
+  echo "Waiting for $MACHINE_CONFIG_POOL MachineConfigPool to start updating..."
+  oc wait mcp $MACHINE_CONFIG_POOL --for='condition=UPDATING=True' --timeout=300s &>/dev/null
+  
+  done
+  
+  for MACHINE_CONFIG_POOL in $KERNEL_MCPS
+  do
+  
+  echo "Waiting for $MACHINE_CONFIG_POOL MachineConfigPool to finish updating..."
+  oc wait mcp $MACHINE_CONFIG_POOL --for='condition=UPDATED=True' --timeout=$KERNEL_TIMEOUT 2>/dev/null
+  
+  done
+}
+
+
+validate_params
+
+create_machine_configs "$INSTALL_KERNEL_MCPS" "$INSTALL_KERNEL_TYPE"
+
+wait_for_config_to_be_applied "$INSTALL_KERNEL_MCPS" "$INSTALL_KERNEL_TIMEOUT"
+
+oc get nodes
+
+for node in $(oc get nodes -o name) 
+do
+  oc debug $node -- chroot /host uname -a
+done
```

Comment:
can we generate a junit in case of failures for this script?

Reply:
@aleskandro please, could you review if what I added is enough? Thank you!

Reply:
@JianLi-RH can you help checking the junit?